lib/memoist.rb

```diff
@@ -63,28 +63,49 @@ def unmemoize_all
       flush_cache
     end
 
+    def init_lazy_memoized_structs(names)
+      structs = self.class.memoized_methods
+      return structs if names.empty?
+
+      structs.select { |s| names.include?(s.memoized_method) }
```

Comment:
If you yield each matching struct here, instead of returning an array, you'll save an allocation